design-proposals/helm-deployment/helmDeploymentRequirements.md

```diff
@@ -0,0 +1,999 @@
+          args:
+            - "--endpoint=$(CSI_ENDPOINT)"
+            - "--controller"
+            - "--containerized"
+            - "--backend-update-interval=120"
+            - "--driver-name=csi.huawei.com"
+          env:
+            - name: CSI_ENDPOINT
+              value: /var/lib/csi/sockets/pluginproxy/csi.sock
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
+          volumeMounts:
+            - name: socket-dir
+              mountPath: /var/lib/csi/sockets/pluginproxy/
+            - name: log
+              mountPath: /var/log
+            - name: config-map
+              mountPath: /etc/huawei
+            - name: secret
+              mountPath: /etc/huawei/secret
+      volumes:
+        - name: socket-dir
+          emptyDir:
+        - name: log
+          hostPath:
+            path: /var/log/
+            type: Directory
+        - name: config-map
+          configMap:
+            name: huawei-csi-configmap
+        - name: secret
+          secret:
+            secretName: huawei-csi-secret
+```
+##### huawei-csi-node.yaml
+Following are kubernetes resources maintained in template file for a node serviced to get deployed:
+- Daemonset
+- ServiceAccount
+- ClusterRole
+- ClusterRoleBinding
+
+Sample:
+```text
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: huawei-csi-node
+  namespace: kube-system
+
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: huawei-csi-driver-registrar-runner
+  namespace: kube-system
+rules:
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["get", "list", "watch", "create", "update", "patch"]
+
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: huawei-csi-driver-registrar-role
+subjects:
+  - kind: ServiceAccount
+    name: huawei-csi-node
+    namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: huawei-csi-driver-registrar-runner
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: huawei-csi-node
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      app: huawei-csi-node
+  template:
+    metadata:
+      labels:
+        app: huawei-csi-node
+    spec:
+      serviceAccountName: huawei-csi-node
+      hostPID: true
+      hostNetwork: true
+      containers:
+        - name: csi-node-driver-registrar
+          image: {{ .Values.images.sidecar.registrar }}
+          args:
+            - "--csi-address=/csi/csi.sock"
+            - "--kubelet-registration-path=/var/lib/kubelet/plugins/csi.huawei.com/csi.sock"
+          imagePullPolicy: {{ .Values.imagePullPolicy }}
+          volumeMounts:
+            - name: plugin-dir
+              mountPath: /csi
+            - name: registration-dir
+              mountPath: /registration
+
+        - name: huawei-csi-driver
+          image: {{ required "Must provide the CSI controller service node image." .Values.images.huaweiCsiNodeService }}
```

Comment:
typo: "Must provide the CSI controller service node image." --> ""Must provide the CSI node service image.""

Reply:
Done